### DISCUSSION TITLE
Output files aren't on actual cuts

### DISCUSSION THREAD

**Discussion by gravelfreeman** [mifi/lossless-cut#1678] [Support]
Hello,

I can't get the app to work right.

No matter how I do my cuts or modify the export settings, I'm never getting the cut on the actual selected key frame. Funny thing is when selecting "chapters only" in the export window, it'll create chapters based on my actual cuts. I've processed that output file, with added chapters, through my mkvmerge script

`"C:\Program Files\MKVToolNix\mkvmerge.exe" -o "!outputDir!!fileName!.mkv" --split chapters:all "!episodeFile!"`

and it successfully separated all my chapters in individual mkv files like I would have expected Lossless Cut to do. So I believe Lossless Cut should be able to get the job done in one pass instead of having to run the exported file through my script.

The second issue I've encountered is with the experimental smart cut feature. It seems to be ignoring my segments and jump straight to the closest key frame. Perhaps AVC codec isn't supported, I've only seen mentions about H264 and H265 but it was from an old issue on this repo. I tried SolveigMM VideoSplitter and I was able to split this same file even though it's AVC codec. That's what made me think it would work on Lossless Cut, but I may be in the wrong.

Thank you for your app and for helping me!

**Comment by mifi** [comment]
Hi. I suspect that you're hitting this issue #1216 - see in particular this workaround: https://github.com/mifi/lossless-cut/issues/330#issuecomment-827200239

Does this problem happen also with other files? (e.g. files from other cameras or sources) - if so, is it possible to share a file that can reproduce the problem?

**Reply by gravelfreeman** [reply]
Thank you for your reply and sorry it took some time to reply back. I'm pretty confident I tried your workaround when I saw that it was always off one frame. Still I seem unable to cut correctly using your app. [As requested I've uploaded an episode to pCloud](https://transfer.pcloud.com/download.html?code=5Z7PDaVZrDd8neKCt97ZAldAZtEWswjVck18xohmSVO1WzHzTOrTX). Please download it asap since the link isn't going to stay up for a long time.

**Reply by mifi** [reply]
I can confirm that it's really hard to get it to cut at a keyframe with your video.
- I tried to cut from the keyframe at 04:41:560 (frame 7039). However it instead exports video beginning at the previous keyframe 04:37.040 (6926).
- If I step forward frame by frame 3 times it still exports from the (same) previous keyframe.
- If  I step one more frame (4 frames, to 7043), then it exports correctly beginning from the keyframe at frame 7039. This resonates with what I discussed in #1216.

What you can try is to run this before export
- Edit -> Segments -> Shift all segments on timeline
- Enter `00:00:00.200` (or a larger value if there's still problems)
- When asked about Start or End timestamps, Select **Start**

This will effectively shift all start times of segments by a little more than 3 frames

**Reply by hrxn** [reply]
Is there a way to identify video files that exhibit those kinds of issues?
Or at least to make something of an educated guess?

**Reply by mifi** [reply]
I haven’t found any way other than exporting and looking at the output

**Reply by hrxn** [reply]
Ah, that's unfortunate. I was hoping that maybe a certain combination of video codec plus container format could be considered a "risk" factor here.
But still, thank you for answering.

**Reply by mifi** [reply]
it might be possible to find some combination but I haven't found one yet.
